### PR TITLE
[W5.7][T09-3] Ong Kai Le

### PR DESCRIPTION
**Pull request by chivent**
added tag command: finds available entries by tag

### CODE REVIEW

**Review by ihwk1996 — COMMENTED** [review]
Looks good! Maybe can add the add testing to input.txt and expected.txt!

**Comment by j-lum** [comment]
I don't see anything wrong with the new feature but you might want to run the build locally and sort out the issues?

**Comment by lancelotzty** [comment]
that is good!!! : D

File: docs/UserGuide.adoc [hunk 1]
```diff
@@ -91,6 +91,23 @@ Returns `John Doe` but not `john`.
...
+and persons matching at least one keyword will be returned (i.e. `OR` search).
+====
+
+Examples:
+
+* `tag friends` +
+Returns people with tag 'friends' or 'Friends' or any other case variation
+
+* `tag friends family` +
+Returns people with tag 'friends', 'family', or both, in any case variation.
```
Comment (eugenepeh) [inline docs/UserGuide.adoc:109]: Good!

File: src/seedu/addressbook/commands/TagCommand.java [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+
+    public TagCommand(Set<String> keywords) {
+        this.keywords = keywords;
+    }
+
+    /**
+     * Returns a copy of keywords in this command.
+     */
+    public Set<String> getKeywords() {
+        return new HashSet<>(keywords);
```
Comment (eugenepeh) [inline src/seedu/addressbook/commands/TagCommand.java:34]: Note that if you might want to return it as an immutable set if its not meant to be modifiable.

File: src/seedu/addressbook/commands/TagCommand.java [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+     */
+    private List<ReadOnlyPerson> getPersonsContainingTag(Set<String> keywords) {
+        final List<ReadOnlyPerson> matchedPersons = new ArrayList<>();
+        for (ReadOnlyPerson person : addressBook.getAllPersons()) {
+            final Set<Tag> tagsOfPerson = person.getTags();
+            for (Tag tag: tagsOfPerson) {
+                for (String input: keywords) {
+                    if (input.equals(tag.tagName.toLowerCase())) {
+                        if (!matchedPersons.contains(person)) {
+                            matchedPersons.add(person);
```
Comment (eugenepeh) [inline src/seedu/addressbook/commands/TagCommand.java:57]: Seems that too much logic in order to get to here, possible to simplify it further?

File: test/java/seedu/addressbook/commands/TagCommandTest.java [hunk 4]
```diff
@@ -0,0 +1,53 @@
+package seedu.addressbook.commands;
+
+import org.junit.Test;
+import seedu.addressbook.data.AddressBook;
+import seedu.addressbook.data.exception.IllegalValueException;
+import seedu.addressbook.data.person.ReadOnlyPerson;
+import seedu.addressbook.util.TypicalPersons;
+
+import java.util.*;
```
Comment (eugenepeh) [inline test/java/seedu/addressbook/commands/TagCommandTest.java:9]: remember to disable intellij auto import wildcard
[Why is wild card import bad?](https://stackoverflow.com/questions/147454/why-is-using-a-wild-card-with-a-java-import-statement-bad)

**Review by eugenepeh — APPROVED** [review]
Good job on completing the activity.
Some comments added. Please close the PR after reading the comments.

**Pull request closed without merging**